sufia-models/lib/sufia/models/stats/user_stat_importer.rb

```diff
@@ -38,8 +38,16 @@ def date_since_last_cache(user)
       end
     end
 
-    def files_for_user(user)
-      ::GenericFile.where(Solrizer.solr_name('depositor', :symbol) => user.user_key)
+    def file_ids_for_user(user)
+      page_size = 1000
+      ids = []
+      resp = ActiveFedora::SolrService.instance.conn.get "select", params:{fq:["#{Solrizer.solr_name("has_model", :symbol)}:\"info:fedora/afmodel:GenericFile\"", "#{Solrizer.solr_name('depositor', :symbol)}:\"#{user.user_key}\""],fl:["id", "#{Solrizer.solr_name('depositor', :symbol)}"], rows: page_size}
+      total = resp["response"]["numFound"]
+      while (ids.length < total) && (resp["response"]["docs"].count > 0)
+        ids.concat resp["response"]["docs"].map{|doc| doc["id"]}
+        resp = ActiveFedora::SolrService.instance.conn.get "select", params:{fq:["#{Solrizer.solr_name("has_model", :symbol)}:\"info:fedora/afmodel:GenericFile\"", "#{Solrizer.solr_name('depositor', :symbol)}:\"#{user.user_key}\""],fl:["id", "#{Solrizer.solr_name('depositor', :symbol)}"], rows: page_size, start: resp["response"]["start"]+page_size}
```

Comment:
except for `start` you're duplicating what's on line 44. Can you store the options in a variable?